azure-cosmos-db/1.0/modules/ROOT/pages/azure-cosmos-db-connector-reference.adoc

```diff
@@ -0,0 +1,1939 @@
+** MILLISECONDS
+** SECONDS
+** MINUTES
+** HOURS
+** DAYS |  Time unit for the *Connection Timeout* field. |  SECONDS |
+| Use Persistent Connections a| Boolean a|  Indicates whether to use persistent connections:
+
+* `true`
++
+Mule uses persistent connections.
+
+* `false`
++
+Mule closes the connection after the first request completes. |  true |
+| Max Connections a| Number |  Maximum number of connections to open to the backend. HTTP requests are sent in parallel over multiple connections. Setting this value too high can impact latency and consume additional resources without increasing throughput. |  -1 |
+| Connection Idle Timeout a| Number |  When persistent connections are enabled, how long a connection can remain idle before Mule closes it. |  30 |
+| Connection Idle Timeout Unit a| Enumeration, one of:
+
+** NANOSECONDS
+** MICROSECONDS
+** MILLISECONDS
+** SECONDS
+** MINUTES
+** HOURS
+** DAYS |  Time unit for the *Connection Idle Timeout* field. |  SECONDS |
+| Stream Response a| Boolean |  If this value is `true`, Mule streams received responses. |  false |
+| Response Buffer Size a| Number |  Size of the buffer that stores the HTTP response, in bytes. |  -1 |
+| Database account url a| String |  Database account URL. |  | x
+| Resource token a| String |  Resource token. |  | x
+| Version a| String |  Version of the Cosmos DB REST service. |  | x
+| TLS configuration a| <<Tls>> |  If HTTPS is configured as a protocol, the user must configure at least the keystore configuration. |  |
+| Reconnection a| <<Reconnection>> a|  What to do if, when an app is deployed, a connectivity test does not pass after exhausting the associated reconnection strategy:
+
+* `true`
++
+Allow the deployment to fail.
+
+* `false`
++
+Ignore the results of the connectivity test. |  |
+|===
+
+== Supported Operations
+* <<CreateDocument>>
+* <<CreateStoredProcedure>>
+* <<DeleteDocument>>
+* <<DeleteStoredProcedure>>
+* <<ExecuteStoredProcedure>>
+* <<GetDocument>>
+* <<ListDocuments>>
+* <<ListStoredProcedures>>
+* <<QueryDocuments>>
+* <<ReplaceDocument>>
+* <<ReplaceStoredProcedure>>
+
+
+
+== Operations
+
+[[CreateDocument]]
+== Create Document
+`<azure-cosmos-db-connector-mule-4:create-document>`
+
+
+Creates a new document in a collection. This operation makes an HTTP POST request to the /dbs/{dbId}/colls/{collId}/docs endpoint.
+
+
+=== Parameters
+
+[%header%autowidth.spread]
+|===
+| Name | Type | Description | Default Value | Required
+| Configuration | String | Name of the configuration to use. | | x
+| Db Id a| String |  User-generated name or ID of the database in which the stored procedure is created. |  | x
+| Coll Id a| String |  Name of the collection in which the stored procedure is created. |  | x
+| X ms documentdb is upsert a| Boolean |  If set to `true`, Cosmos DB creates the document with the ID (and partition key value if applicable) if the document doesn't exist, or updates the document if the document does exist. |  false |
```

Comment:
If set to `true`, Cosmos DB  does the following:

* Creates the document with the ID (and partition key value if applicable), if the document doesn't exist
* Updates the document, if it exists.